ansible/upgrade_sonic.yml

```diff
@@ -61,15 +61,18 @@
             disk_used_pcent: '{{disk_used_pcent}}'
             new_image_url: '{{ image_url }}'
 
+        # Reboot may need some time to update firmware firstly.
+        # Increasing the async time to 300 seconds to avoid reboot being interrupted.
         - name: Reboot the switch {{ inventory_hostname }}
           become: true
           shell: reboot
-          async: 1
+          async: 300
```

Comment:
adding hard coded 5 minutes should be a workaround.
if we believe in some cases more than 1 sec is required we should define minimal time and retry mechanism

Reply:
The async value specifies the task's maximum runtime.  And the task is executed asynchronously. It does not block execution of subsequent tasks. It can't be too short because the reboot command may need to update firmware firstly. Another thing special about the reboot command is that when it's execution is done, it also shutdown the system for rebooting. So, polling its execution status may fail eventually with lose connection to device. The full working flow is: 
* Issue the reboot command, give it enough time for execution. Move on to the next task immediately. 
* The next task is to wait for device to come back. It delays 180 seconds and then starts checking SSH connection to device. If device is not back, then retry until timeout.